modules/widgets/eu-cookie-law.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * Disable direct access/execution to/of the widget code.
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class Jetpack_EU_Cookie_Law_Widget extends WP_Widget {
+	public static $cookie_name = 'eucookielaw';
+	public static $cookie_validity = 2592000; // 30 days
+
+	public $defaults = array(), $instance;
+
+	function __construct() {
+		parent::__construct(
+			'eu_cookie_law_widget',
+			apply_filters( 'jetpack_widget_name', esc_html__( 'EU Cookie Law Banner', 'jetpack' ) ),
+			array(
+				'description' => esc_html__( 'Display a banner for compliance with the EU Cookie Law.', 'jetpack' ),
+			),
+			array()
+		);
+
+		$this->defaults = array(
+			'hide'               => 'button',
+			'hide-timeout'       => 30,
+			'text'               => 'default',
+			'customtext'         => '',
+			'color-scheme'       => 'default',
+			'policy-url'         => 'default',
+			'default-policy-url' => 'https://en.support.wordpress.com/cookies',
+			'custom-policy-url'  => '',
+			'policy-link-text'   => __( 'Our Cookie Policy', 'jetpack' ),
+			'button'             => __( 'Close and accept', 'jetpack' ),
+			'default-text'       => __( 'Privacy & Cookies: This site uses cookies from WordPress.com and selected partners. ', 'jetpack' ),
+		);
+
+		if ( is_active_widget( false, false, $this->id_base ) || is_customize_preview() ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_style' ) );
+		}
+	}
+
+	public function widget( $args, $instance ) {
+		$this->instance = wp_parse_args( $instance, $this->defaults );
+
+		do_action( 'jetpack_stats_extra', 'widget_view', 'eu_cookie_law' );
+
+		add_action( 'wp_footer', array( $this, 'footer' ) );
+	}
+
+	public function footer() {
+		$blog_url = get_bloginfo( 'url' );
+		$instance = $this->instance;
+		$defaults = $this->defaults;
+		$cookie_name = self::$cookie_name;
+		$cookie_validity = self::$cookie_validity;
+
+		require( dirname( __FILE__ ) . '/eu-cookie-law/footer.php' );
+	}
+
+	public function form( $instance ) {
```

Comment:
It would be a nice plus to add docblocks before each one of your functions. If you install `phpcs`, you'll see that it complains when no comments are added:
https://github.com/Automattic/jetpack/blob/master/docs/coding-guidelines.md#general